### PR TITLE
add docker integration tests

### PR DESCRIPTION
**Pull request by talevy**

### CODE REVIEW

File: spec/integration/outputs/kafka_spec.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+describe "outputs/kafka", :integration => true do
+  let(:test_topic) { 'test' }
+  let(:config) { {'client_id' => 'spectest', 'bootstrap_servers' => "#{@kafka_host}:#{@kafka_port}", 'topic_id' => test_topic} }
+  let(:event) { LogStash::Event.new({'message' => 'hello', 'topic_name' => 'my_topic', '@timestamp' => LogStash::Timestamp.now}) }
+
+  context 'when outputting messages' do
+    let(:num_events) { 3 }
+
+    it 'should send logstash event to kafka broker' do
```
Comment (andrewvc) [inline spec/integration/outputs/kafka_spec.rb]: s/event/events/

File: spec/integration/outputs/kafka_spec.rb [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      kafka = LogStash::Outputs::Kafka.new(config)
+      kafka.register
+      num_events.times do kafka.receive(event) end
+      kafka.teardown
+
+      consumer = Poseidon::PartitionConsumer.new("my_test_consumer", @kafka_host, @kafka_port, test_topic, 0, :earliest_offset)
+      messages = consumer.fetch
+      expect(messages.size).to eq(num_events)
```
Comment (andrewvc) [inline spec/integration/outputs/kafka_spec.rb]: It'd be nice to make the messages the subject and have two specs, one for message count, one for message value.

**Comment by andrewvc** [comment]
Looks good save a couple style things I commented on

**Comment by suyograo** [comment]
Nice, would love to see more tests for `message_key` (that it ends up in the right topic) and `value_serializer`.. maybe use avro/kryo to persist messages and use a consumer on the other end to deserialize

File: spec/integration/outputs/kafka_spec.rb (hunk 2 above)
Comment (talevy) [inline spec/integration/outputs/kafka_spec.rb]: agreed. updated to reflect this setup

**Comment by talevy** [comment]
update: added test for `message_key`, wherein all messages should be sent to a singular partition for a topic that contains two total partitions.

**Comment by suyograo** [comment]
LGTM

**Pull request closed without merging**